include/deal.II/lac/vector.templates.h

```diff
@@ -120,23 +120,32 @@ namespace internal
 
   template <typename Functor>
   void vectorized_transform(Functor &functor,
-                            size_type vec_size)
+                            size_type vec_size,
+                            std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner)
   {
 #ifdef DEAL_II_WITH_THREADS
     // only go to the parallel function in case there are at least 4 parallel
     // items, otherwise the overhead is too large
     if (vec_size >= 4*internal::Vector::minimum_parallel_grain_size &&
         MultithreadInfo::n_threads() > 1)
       {
+        if (partitioner.get() == NULL)
+          partitioner.reset(new parallel::internal::TBBPartitioner());
```

Comment:
How would this happen? Or is it the case that the vector does not by default allocate a partitioner and the first time we get here the allocation actually happens?

I ask because this place presents a race condition. Nothing wrong will happen, but two threads may still race to set the partitioner to an object both threads acquire between the `if` and the `reset` call; in the end, the shared pointer class accepts both and happens to throw away the first of the two, but you may get into trouble if the first of these threads already proceeds to the `acquire_one_partitioner` call below before the `partitioner` variable is reset by the second thread.